src/Form/SettingsForm.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\Entity\ConfigEntityStorageInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Settings form.
+ */
+class SettingsForm extends ConfigFormBase {
+
+  const CONSTRAINT_EFFECT_STRIKETHROUGH = 'strikethrough';
+  const CONSTRAINT_EFFECT_HIDE = 'hide';
+
+  /**
+   * Default minimum password length.
+   */
+  const MINIMUM_LENGTH = 8;
+
+  /**
+   * Password constraint entity storage.
+   *
+   * @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface
+   */
+  protected $passwordConstraintEntityStorage;
+
+  /**
+   * Constructs the settings from for password constraint.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Config factory.
+   * @param \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $password_constraint_entity_storage
+   *   Password constraint entity storage.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Messenger.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, ConfigEntityStorageInterface $password_constraint_entity_storage, MessengerInterface $messenger) {
+    parent::__construct($config_factory);
+
+    $this->passwordConstraintEntityStorage = $password_constraint_entity_storage;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): SettingsForm {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('entity_type.manager')->getStorage('password_enhancements_constraint'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'password_enhancements_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames(): array {
+    return [
+      'password_enhancements.settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    if ($this->config('user.settings')->get('password_strength')) {
+      $this->messenger->addWarning($this->t('The core <a href=":url" target="_blank">password strength indicator</a> is enabled, it is recommended to disable it.', [
```

Comment:
I would rather say "The built-in ..." because it could be more meaningful to even less technical ppl.